src/desktop/paths.cpp

```diff
@@ -106,7 +106,18 @@ void enumerate_storage_devices(std::vector<path_info>& res)
 	// reasoning here is that if any or all of them are non-empty, they are
 	// probably used for _something_ that might be of interest to the user (if not
 	// directly and actively controlled by the user themselves).
-	static const std::vector<std::string> candidates { "/media", "/mnt" };
+	// Note that the first candidate is actually /run/media/USERNAME -- we need
+	// to fetch the username at runtime from passwd to complete the path.
+	std::vector<std::string> candidates { "/run/media/", "/media", "/mnt" };
+
+	// Fetch passwd entry for the effective user the current process runs as
+	if(const passwd* pw = getpwuid(geteuid()); pw) {
```

Comment:
If you're going to go to the trouble of using the semicolon here, I'd think I'd prefer to write out a proper expression (`pw != nullptr`). Or just drop the `; pw` altogether.